src/data/repositories/OptionsD2Repository.ts

```diff
@@ -24,28 +24,42 @@ export class OptionsD2Repository implements OptionsRepository {
             });
     }
 
-    getAllHazardTypes(): FutureData<Option[]> {
+    getDataSources(): FutureData<Option[]> {
+        return this.getOptionSetByCode("RTSL_ZEB_OS_DATA_SOURCE");
+    }
+
+    getHazardTypes(): FutureData<Option[]> {
         return this.getOptionSetByCode("RTSL_ZEB_OS_HAZARD_TYPE").map(hazardTypes => {
-            return hazardTypes.map(hazardType => ({
-                id: getHazardTypeByCode(hazardType.id),
-                name: hazardType.name,
-            }));
+            return hazardTypes.reduce((acc: Option[], hazardType: NamedRef) => {
+                const hazardTypeId = getHazardTypeByCode(hazardType.id);
+                if (hazardTypeId) {
+                    return [
+                        ...acc,
+                        {
+                            id: hazardTypeId,
+                            name: hazardType.name,
+                        },
+                    ];
+                } else {
+                    return acc;
+                }
+            }, [] as Option[]);
```

Comment:
The usage of `reduce` makes sense only for very custom loops that have some interaction with the accumulated value.  There's usually a more declarative pattern. In this case: map + return undefined/null if nothing to map + compact.